File: r2.py
import numpy as np
arr = np.zeros((33,33,32))
iss = np.zeros((32,2,16))
def func(selling_team,purchasing_team,gun_code,gun_quantity,gun_price):

  if((selling_team<31 and selling_team>0)and(purchasing_team<31 and purchasing_team>0)and(gun_code>0 and gun_code<16)and(gun_quantity>0)and(gun_price>0)):
    arr[selling_team][31][0] = (gun_quantity*gun_price)
    arr[selling_team][32][0] = arr[selling_team][32][0] + arr[selling_team][31][0]
    arr[31][purchasing_team][0] = (gun_quantity*gun_price)
    arr[32][purchasing_team][0] = arr[32][purchasing_team][0] - arr[31][purchasing_team][0]
    arr[32][selling_team][0] = arr[selling_team][32][0]
    arr[purchasing_team][32][0] = arr[32][purchasing_team][0]
    iss[selling_team][0][gun_code] = iss[selling_team][0][gun_code] - gun_quantity
    iss[purchasing_team][0][gun_code] = iss[purchasing_team][0][gun_code] + gun_quantity
  else:
    print("Error in the given data maybe exceed in matrix limit\n")

print("Update gun matrix!!\n")
for i in range(1,6):
  print("Update team {i} gun_quantity and new balance\n".format(i=i))
  gcc = int(input("Update team net balance : "))
  iss[i][1][0] = gcc
  nog = int(input("Number of guns purchased : "))
  for j in range(1,nog+1):
      gcc = int(input("Gun code for team {i} : ".format(i=i)))
      gqq = int(input("Gun Quantity for gun code {gcc} for team {i} : ".format(gcc=gcc,i=j)))
      iss[i][0][gcc] = gqq
for i in range(1,32):
  arr[32][i][0] = iss[i][1][0]
  arr[i][32][0] = iss[i][1][0]
for i in range(1,32):
  print("{aa}\t{bb}\n".format(aa = arr[i][32][0],bb = arr[32][i][0]))
while(1):
  n = int(input("password/n"))
  if n==53555355 :
      print("Terminating\n")
      print("DATA Output of balance\n")
      for i in range(1,32):
          print("Team {i} Net Selling balance {bb} and Net Purchasing balance {cc} And Guns\n".format(i=i,bb=arr[i][32][0],cc=arr[32][i][0]))
          for j in range(1,16):
            print(" Gun Code {i} : {dd}/n".format(i=i,dd=iss[i][0][j]))
      break
  else:
    print("New update\n")
    s = int(input("Selling_team "))
    p = int(input("Purchasing_team "))
    gc = int(input("Gun_code "))
    gq = int(input("Gun_Quantity "))
    gp = int(input("Gun_Price "))
    func(s,p,gc,gq,gp)
    print("******UPDATED******")

File: r2.cpp
#include<iostream>
using namespace std;
int arr[33][33][32];
int iss[31][16];
void func(int selling_team,int purchasing_team,int gun_code,int gun_quantity,int gun_price)
{
  arr[selling_team][purchasing_team][2*gun_code] = gun_quantity;
  arr[selling_team][purchasing_team][2*gun_code+1] = gun_price;
  arr[selling_team][31][0] = (gun_quantity*gun_price);
  arr[selling_team][32][0] = arr[selling_team][32][0] + arr[selling_team][31][0];
  arr[31][purchasing_team][0] = (gun_quantity*gun_price);
  arr[32][purchasing_team][0] = arr[32][purchasing_team][0] - arr[31][purchasing_team][0];
  arr[32][selling_team][0] = arr[selling_team][32][0];
  arr[purchasing_team][32][0] = arr[32][purchasing_team][0];
  iss[selling_team][gun_code] = iss[selling_team][gun_code] - gun_quantity;
  iss[purchasing_team][gun_code] = iss[purchasing_team][gun_code] + gun_quantity;
}
int main()
{
  for(int i = 0;i<31;i++)
  {
    arr[i][0][0] = i+1;
  //Initial cost to be given
    arr[32][i][0] = 0;
    arr[i][32][0] = 0;
  }
  for(int i = 0;i<31;i++)
  {
    arr[0][i][0] = i+1;
  }
  for(int i = 0;i<=31;i+=2)
  {
    arr[0][0][i] = i+1;
  }
  cout<<"Update gun matrix!!"<<endl;
  for(int i = 1;i<31;i++)
  {
    int gcc;
    cout<<"Update team "<<i<<" gun_quantity"<<endl;
    for(int j = 1;j<16;j++)
    {
      cout<<"Gun Quantity for gun code "<<j<<" is : ";
      cin>>gcc;
    }
  }
  while(1)
  {
    int n,s,p,gc,gq,gp;
    cout<<"password"<<endl;
    cin>>n;
    if(n==53555355)
      {
        cout<<"Terminating"<<endl;
        for(int i = 0;i<3)
        break;
      }
    else
    {
      cout<<"New update "<<endl;
      cout<<"Selling_team"<<endl;
      cin>>s;
      cout<<"purchasing_team"<<endl;
      cin>>p;
      cout<<"Gun_code"<<endl;
      cin>>gc;
      cout<<"Gun_quantity"<<endl;
      cin>>gq;
      cout<<"Gun_price"<<endl;
      cin>>gp;
      func(s,p,gc,gq,gp);
      cout<<"******UPDATED******\n\n";
    }
  }
  return 0;
}
